lib/_http_server.js

```diff
@@ -189,7 +189,7 @@ ServerResponse.prototype.writeHead = function(statusCode, reason, obj) {
 
   statusCode |= 0;
   if (statusCode < 100 || statusCode > 999)
-    throw new RangeError(`Invalid status code: ${statusCode}`);
+    throw new RangeError(`Invalid status code: ${this.statusCode}`);
```

Comment:
I'd prefer not. The test that leads to this error is being performed on the coerced value. I would rather display the coerced value than the actual value.